test/gun_SUITE.erl

```diff
@@ -107,6 +107,21 @@ ignore_empty_data_http(_) ->
 	1 = length(Zero),
 	gun:close(Pid).
 
+ignore_empty_data_fin_http(_) ->
+	doc("When gun:data/4 is called with fin and empty data, it must be ignored."),
+	{ok, OriginPid, OriginPort} = init_origin(tcp, http),
+	{ok, Pid} = gun:open("localhost", OriginPort),
+	{ok, http} = gun:await_up(Pid),
+	handshake_completed = receive_from(OriginPid),
+	Ref = gun:post(Pid, "/", []),
+	gun:data(Pid, Ref, nofin, "hello"),
+	gun:data(Pid, Ref, fin, ""),
+	Data = receive_all_from(OriginPid, 500),
+	Lines = binary:split(Data, <<"\r\n">>, [global]),
+	Zero = [Z || <<"0">> = Z <- Lines],
+	1 = length(Zero),
```

Comment:
Took me a while to figure out you were looking for 0\r\n. Maybe search for `<<"\r\n0\r\n">>` via `binary:match` instead.

Reply:
I copied the existing test case `ignore_empty_data_http`.
W/o the fix, the code sent two "0".  This test ensures it just gets one.

Reply:
Even better if you can fix that test as well.

Reply:
Perhaps instead of doing this it would be better to split in two at `\r\n\r\n` and try to parse the body.